api/db/populate_users.js

```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env node
+
+// TODO: This will all be in the general js to abide by DRY principal
+const userArgs = process.argv.slice(2);
+
+if (!userArgs[0].startsWith('mongodb')) {
+  console.log(
+    'Error: you need to specify a valid MongoDB URL as the first argument'
+  );
+
+  return;
+}
+
+const User = require('../models/User');
+const Interest = require('../models/Interest')
+const Token = require('../models/Token')
+const crypto = require('crypto');
+
+const async = require('async');
+
+const mongoose = require('mongoose');
+const mongoDB = userArgs[0];
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = global.Promise;
+const db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+const { uniqueNamesGenerator, names } = require('unique-names-generator');
+const countries = require('country-region-data');
+const locales = ['en', 'es'];
+const users = [];
+
+const createUser = async (
+  name, email, emailConfirmed, emailConfirmedAt,
+    dateOfBirth, country, region,
+    preferredLocale, interests, admin,
+    callback
+) => {
+  const user = new User({
+    name, email, emailConfirmed, emailConfirmedAt,
+    dateOfBirth, country, region,
+    preferredLocale, interests, admin
+  });
+  user.setPassword('Password1');
+  await user.save();
+  const token = createTokenForUser(user);
+  await token.save();
+
+  console.log(`User: ${user}`);
+  users.push(user);
+  callback(null, user);
+}
+
+function randomDate(start, end) {
+  return new Date(+start + Math.random() * (end - start));
+}
+
+createTokenForUser = (user) => {
+  return new Token({
+    user,
+    type: 'Email',
+    token: crypto.randomBytes(16).toString('hex')
+  });
+}
+
+const generateUsersArray = async () => {
+  let interests = await Interest.distinct("_id");
+  const config = {
+    dictionaries: [names]
+  }
+
+  let arr = []
+  for (let i = 0; i < 100; i++) {
+    const name = `${uniqueNamesGenerator(config)} ${uniqueNamesGenerator(config)}`;
+    const email = `${name.replace(/ /g, '_')}@test-tisn-email.com`.toLowerCase();
+    const country = countries[Math.floor(Math.random() * countries.length)];
+    const region = country.regions[Math.floor(Math.random() * country.regions.length)];
+    const enrolledDate = randomDate(new Date(2019, 01, 01), new Date());
+    const birthDate = randomDate(new Date(1950, 01, 01), new Date(2007, 01, 01));
+    interests = interests.sort(() => 0.5 - Math.random());
+    const selectedInterests = interests.slice(0, Math.floor(Math.random() * 10));
+
+
+    arr.push(
+      (seriesCallback) => {
+        createUser(
+          name,
+          email,
+          true,
+          enrolledDate,
+          birthDate,
+          country.countryShortCode,
+          region.shortCode || 'N/A',
+          locales[Math.floor(Math.random() * 2)],
+          selectedInterests,
+          false,
```

Comment:
How would you like that to be handled? I can make it a flag to be passed in the command line. Or I can have 1 (or any number) admin created each time.

Reply:
I think I would just create one admin user and document its existence so developers know of it and how to use it. Also, maybe it's a good idea to use some very clear values for it: `Admin` as its name, `admin@tisn.app` as its email, Tisn's logo as its avatar, etc. The last example just reminded me of something: should we try and use random images from somewhere for the dummy users' avatars, or is that an overkill?

Reply:
I don't think it is overkill, though unless it is something super simple it might be worth opening a feature issue for it. 

As for logging to the user, it might be worth looking at a better way of managing and displaying important information to the user. Maybe by default we don't display anything but pertinent information (number of records created, admin information, etc.). And then we can add an optional command line flag for displaying all logs.

Reply:
For the avatars, we can hopefully use some service like [Adorable Avatars](http://avatars.adorable.io/) quite easily! That's the first thing I found when googling, I'm just dropping the link here.

Reply:
Awesome I will make note of it and look into it.